datalad/crawler/nodes/annex.py

```diff
@@ -336,7 +336,7 @@ def __init__(self, path=None,
         self._providers = Providers.from_config_files()
         self.yield_non_updated = yield_non_updated
 
-        if (not allow_dirty) and self.repo.dirty:
+        if (not allow_dirty) and self.repo.dirty():
```

Comment:
The point is to make it clear to the user, that there are parameters and that it is not necessarily obvious, what to consider dirty. Having a property suggests there is a notion of "dirty" that always makes sense and is the same for an annex repo and a git repo independent of the mode of operation. And this is a troublesome suggestion, which is not true.

In that sense, "all the other similar methods" actually are not that similar. Their meaning doesn't change, there simply are no parameters to tune, what these properties mean or are supposed to return.

However, I agree on renaming to `is_dirty()` to fit the pattern. If you need a shortcut property, pointing to the default values in order to save some bytes - well, feel free to build your one-liner. But I bet this will be used in an invalid way and lead to complaining on how this is confusing ;-)